packages/wp-now/src/pool.ts

```diff
@@ -0,0 +1,180 @@
+import { NodePHP } from '@php-wasm/node';
+
+const Fatal = Symbol('Fatal');
+const Spawn = Symbol('Spawn');
+const Reap = Symbol('Reap');
+
+let childCount = 0;
+
+export class PoolInfo {
+	id = childCount++;
+	requests = 0;
+	started = Date.now();
+	active = false;
+}
+
+/**
+ * Maintains and refreshes a list of php instances
+ * such that each one will only be fed X number of requests
+ * before being discarded and replaced.
+ */
+export class Pool {
+	instances = new Map(); // php => PoolInfo
+
+	spawner: () => Promise<any>; // Callback to create new instances.
+	maxRequests: number; // Max requests to feed each instance
+	maxJobs: number; // Max number of instances to maintain at once.
+
+	notifiers = new Map(); // Inverted promises to notify async code of backlogged item processed.
+	running = new Set(); // Set of busy PHP instances.
+	backlog = []; // Set of request callbacks waiting to be run.
+
+	constructor({
+		spawner = async (): Promise<any> => {},
+		maxRequests = 2000,
+		maxJobs = 5,
+	} = {}) {
+		Object.assign(this, { spawner, maxRequests, maxJobs });
```

Comment:
this is a style note, but I think we would generally prefer to set these like `this.spawner = spawner`

did you see this pattern elsewhere here?

Reply:
That's just something I use as shorthand for multiple assignments.

Reply:
@dmsnell This has been refactored: https://github.com/WordPress/playground-tools/pull/110/files#diff-95251f05603f485be0377cd57ce90fe740821ca840c6053a115cb48c8725341bR156